kuksa-client/kuksa_client/grpc/__init__.py

```diff
@@ -315,6 +315,27 @@ def from_message(cls, message: types_pb2.Datapoint):
             ) if message.HasField('timestamp') else None,
         )
 
+
+    def cast_array_values(cast, array):
+    
+        array = array.strip('[]')
+        pattern = r'(?:\\"|[^",])+|"(?:\\"|[^"])*"'
```

Comment:
This will lead from 
```
setValue Vehicle.OBD.DTCList ["dtc1, dtc2", dtc3, \" dtc4, dtc4\"]
```
to 
```
{
    "path": "Vehicle.OBD.DTCList",
    "value": {
        "value": {
            "values": [
                "dtc1, dtc2",
                "dtc3",
                "\" dtc4",
                "dtc4\""
            ]
        },
        "timestamp": "2023-07-11T06:37:13.441133+00:00"
    }
}
```
the comma in \" will not be treated correct.

Reply:
I am adding some recommendations to the README. Main problem is that not surrounding the whole array with quotes leaves you to the special handling of the shell/argparse. Like in this case (blank before first double quote)

`setValue Vehicle.OBD.DTCList [ "dtc1, dtc2", dtc3, \" dtc4, dtc4\"]`

... which neither now or before give you what you want as the shell "eats" the quotes. We could possibly do something about it, like add quotes back but that gives us a lot of corner cases where we want to escape any quotes inside, but not quotes that already are escaped. So for now we need to live with that limitation. Better then to use surround everything with single quotes:

`setValue Vehicle.OBD.DTCList '[ "dtc1, dtc2", dtc3,   \" dtc4, dtc4\"]'`

Which gives you what i think is correct
```
Test Client> getValue Vehicle.OBD.DTCList                                           
{
    "path": "Vehicle.OBD.DTCList",
    "value": {
        "value": {
            "values": [
                "dtc1, dtc2",
                "dtc3",
                "\" dtc4",
                "dtc4\""
            ]
        },
        "timestamp": "2023-07-12T06:19:14.369683+00:00"
    }
}

```